### PR TITLE
Remove unused variables from coordinates sub-package

### PR DESCRIPTION
**Pull request by ecalifornica**
### Description

This pull request is to address removing unused variables from the `coordinates` sub-package, resolving the linter error [F841](https://docs.astral.sh/ruff/rules/unused-variable/). I attempted to keep it simple and not remove any function calls, since I'm not familiar with the tests. There is likely more to clean up.

I updated the temporary config `.ruff.toml`, but did not update `.flake8`.

Fixes #<Issue Number> N/A
These changes were inspired by:
https://github.com/astropy/astropy/issues/14818
https://github.com/astropy/astropy/issues/8470

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/coordinates/tests/test_frames_with_velocity.py [hunk 1]
```diff
@@ -272,14 +272,12 @@ def test_shorthand_attributes():
         pm_dec=np.random.normal(0, 100, n) * u.mas / u.yr,
         radial_velocity=np.random.normal(0, 100, n) * u.km / u.s,
     )
-    v = icrs1.velocity
```
Comment (neutrinoceros) [inline astropy/coordinates/tests/test_frames_with_velocity.py:275]: this test is about checking that attributes are accessible, so this line shouldn't be removed. One way to keep linters happy while still exercising this functionality is to prefix the variable name as
```python
    _v = icrs1.velocity
```

File: astropy/coordinates/tests/test_frames_with_velocity.py [hunk 2]
```diff
@@ -272,14 +272,12 @@ def test_shorthand_attributes():
...
     pm = icrs1.proper_motion
     assert quantity_allclose(pm[0], icrs1.pm_ra_cosdec)
     assert quantity_allclose(pm[1], icrs1.pm_dec)
 
     # for scalar data:
     icrs2 = ICRS(**POSITION_ON_SKY, **DISTANCE, **PROPER_MOTION, **RADIAL_VELOCITY)
-    v = icrs2.velocity
```
Comment (neutrinoceros) [inline astropy/coordinates/tests/test_frames_with_velocity.py:282]: same as above

File: astropy/coordinates/tests/test_sky_coord.py [hunk 3]
```diff
@@ -1398,8 +1398,8 @@ def test_frame_attr_transform_inherit():
 
 def test_deepcopy():
     c1 = SkyCoord(1 * u.deg, 2 * u.deg)
-    c2 = copy.copy(c1)
-    c3 = copy.deepcopy(c1)
```
Comment (neutrinoceros) [inline astropy/coordinates/tests/test_sky_coord.py:1402]: we could also rename `c4` and `c5` to `c2` and `c3`

File: astropy/coordinates/tests/test_frames_with_velocity.py (hunk 1 above)
Comment (ecalifornica) [reply astropy/coordinates/tests/test_frames_with_velocity.py:275]: Doh! Thanks for catching this, and for teaching me to placate the linter with an underscore, I didn't know that was possible.

File: astropy/coordinates/tests/test_frames_with_velocity.py (hunk 2 above)
Comment (ecalifornica) [reply astropy/coordinates/tests/test_frames_with_velocity.py:282]: Fixed.

File: astropy/coordinates/tests/test_sky_coord.py (hunk 3 above)
Comment (ecalifornica) [reply astropy/coordinates/tests/test_sky_coord.py:1402]: Updated.

**Review by eerovaher — COMMENTED** [review]
Some of the changes this pull request makes are clear improvements, but in many other cases it would be much better to consider rewriting entire tests instead of slightly editing a couple of lines.

File: astropy/coordinates/tests/test_angles.py [hunk 4]
```diff
@@ -459,7 +459,7 @@ def test_radec():
     # dec = Latitude("-41:08:15.162342")
     with pytest.raises(u.UnitsError):
         Latitude("-41:08:15.162342")
-    dec = Latitude("-41:08:15.162342", unit=u.degree)  # same as above
+    Latitude("-41:08:15.162342", unit=u.degree)  # same as above
```
[the remark this answers is not in the document]
Comment (ecalifornica) [reply astropy/coordinates/tests/test_angles.py]: I ran the tests and inserted the output of the Latitude degree attribute, does that work?

File: astropy/coordinates/tests/test_frames_with_velocity.py (hunk 1 above)
Comment (ecalifornica) [reply astropy/coordinates/tests/test_frames_with_velocity.py:275]: I added `hypothesis` tests mimicking what was there before, let me know what you think. I don't have enough domain knowledge yet to feel confident in rewriting the entire test.

File: astropy/coordinates/tests/test_pickle.py [hunk 5]
```diff
@@ -15,7 +15,7 @@
 def test_basic():
     lon1 = Longitude(1.23, "radian", wrap_angle="180d")
     s = pickle.dumps(lon1)
-    lon2 = pickle.loads(s)
+    pickle.loads(s)
```
[the remark this answers is not in the document]
Comment (ecalifornica) [reply astropy/coordinates/tests/test_pickle.py]: Added a naive assertion.

File: astropy/coordinates/tests/test_sky_coord.py [hunk 6]
```diff
@@ -1410,12 +1410,12 @@ def test_deepcopy():
...
-    assert c5.frame.name == c4.frame.name
-    assert c5.obstime == c4.obstime
-    assert c5.equinox == c4.equinox
-    assert c5.representation_type == c4.representation_type
+    c3 = copy.deepcopy(c2)
+    assert np.all(c3.x == c2.x)  # and y and z
+    assert c3.frame.name == c2.frame.name
+    assert c3.obstime == c2.obstime
+    assert c3.equinox == c2.equinox
+    assert c3.representation_type == c2.representation_type
```
[the remark this answers is not in the document]
Comment (ecalifornica) [reply astropy/coordinates/tests/test_sky_coord.py]: Updated the names and added your suggested assertion.

File: astropy/coordinates/tests/test_angles.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (eerovaher) [reply astropy/coordinates/tests/test_angles.py]: Yes, that's good.

File: astropy/coordinates/tests/test_frames_with_velocity.py (hunk 1 above)
Comment (eerovaher) [reply astropy/coordinates/tests/test_frames_with_velocity.py:275]: I'm not sure if this test needs to be implemented as a property-based test, but if it is to be one then the updated code is much better than the original.

File: astropy/coordinates/tests/test_sky_coord.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (eerovaher) [reply astropy/coordinates/tests/test_sky_coord.py]: You've added one `assert ... is not ...` check, but we should be checking more.

**Review by eerovaher — COMMENTED** [review]
I'll explain why I am requesting changes that are going quite far beyond just enforcing `F841` in `coordinates`. Code that is violating some Ruff rule might benefit from being rewritten. For example, it might be possible to simplify code by making use of features that are provided by modern Python or some dependency and which were not available when the code was written. However, if we only ever make minimal changes to address Ruff rules then we might end up with code that could still benefit from being rewritten, but which can't be found easily because Ruff wouldn't highlight it anymore. That is why I think it is good to be quite aggressive when addressing the Ruff rules.

**Comment by ecalifornica** [comment]
> I'll explain why I am requesting changes that are going quite far beyond just enforcing `F841` in `coordinates`. Code that is violating some Ruff rule might benefit from being rewritten. For example, it might be possible to simplify code by making use of features that are provided by modern Python or some dependency and which were not available when the code was written. However, if we only ever make minimal changes to address Ruff rules then we might end up with code that could still benefit from being rewritten, but which can't be found easily because Ruff wouldn't highlight it anymore. That is why I think it is good to be quite aggressive when addressing the Ruff rules.

Makes total sense. For the stickier changes do you think it would be better to revert them and leave them to error (and adding this sub-package to the temporary `ruff` ignore), or add line-specific ignore pragmas?

**Comment by eerovaher** [comment]
This pull request already addresses all violations of `F841` in `coordinates` and I don't see why we would want to revert any of those changes.

**Pull request closed without merging**